ambry-server/src/main/java/com.github.ambry.server/StatsManager.java

```diff
@@ -166,6 +165,28 @@ StatsSnapshot fetchSnapshot(PartitionId partitionId, List<String> unreachableSto
     return statsSnapshot;
   }
 
+  /**
+   * Add new {@link com.github.ambry.store.BlobStore} to StatsManager and thus stats of this store will be collected during
+   * next round of aggregation.
+   * @param id the {@link ReplicaId} associated with store to be added
+   * @return {@code true} if adding store was successful. {@code false} if not.
+   */
+  boolean addBlobStore(ReplicaId id) {
+    if (partitionToReplicaMap.containsKey(id.getPartitionId())) {
+      return false;
+    }
+    partitionToReplicaMap.put(id.getPartitionId(), id);
+    return true;
+  }
+
+  /**
+   * Remove {@link com.github.ambry.store.BlobStore} from StatsManager and thus it won't be checked during next round of aggregation.
+   * @param id the {@link ReplicaId} associated with store to be removed
+   */
+  void removeBlobStore(ReplicaId id) {
```

Comment:
Summary from offline discussion: consider the possibility to check the partitionToReplicaMap at the end of stats collection to remove any unreachable replica that no longer exist in the map. This can help with the scenario when a store is removed while stats collection is running and reporting it as unreachable when it's actually being removed.

Reply:
I need to correct the statement I made during offline discussion.  I misunderstood the description in Oracle docs:
```
Iterators and Enumerations return elements reflecting the state of the hash table at some point at or since the creation of the iterator/enumeration
```
This doesn't mean the iterator is an immutable view of map. Based on my experiments, the iterator would access new added element in concurrent case. So, I took @vgkholla's advice and created new Set as "snapshot" of map's keySet(). The relevant results are presented in concurrent unit tests.